### DISCUSSION TITLE
[ConversationSummaryPlugin][OpenAI] Track Your Token Usage and Costs using OpenAI in ConversationSummaryPlugin

### DISCUSSION THREAD

**Discussion by maivanchuong95** [microsoft/semantic-kernel#4921] [Q&A] [answered]
Hi everyone, 
I have used OpenAI to create Summarization, Actions, and Topics of a paragraph I entered using Semantic Kenel, but currently, I do not know how to calculate the usage cost and how to count the number of tokens for each use. I hope everyone can help me find a solution. Below is my code in .NET, thanks.
My code in base: https://github.com/microsoft/semantic-kernel/blob/main/dotnet/samples/KernelSyntaxExamples/Example13_ConversationSummaryPlugin.cs
My code: `public async Task<string> SummaryPlugin(string context)
{
    Console.WriteLine($"Receive user summary plugin request");

    Kernel kernel = InitializeKernel();

    KernelPlugin conversationSummaryPlugin = kernel.ImportPluginFromType<ConversationSummaryPlugin>();

    FunctionResult summary = await kernel.InvokeAsync(
        conversationSummaryPlugin["SummarizeConversation"], new() { ["input"] = context });
    var responseMessage = summary.GetValue<string>();

    Console.WriteLine($"Response system message: {responseMessage}");

    return responseMessage?? String.Empty;
}

public async Task<string> ActionPlugin(string context)
{
    Console.WriteLine($"Receive user action plugin request");

    Kernel kernel = InitializeKernel();

    KernelPlugin conversationSummary = kernel.ImportPluginFromType<ConversationSummaryPlugin>();

    FunctionResult summary = await kernel.InvokeAsync(
        conversationSummary["GetConversationActionItems"], new() { ["input"] = context });
    var responseMessage = summary.GetValue<string>();

    Console.WriteLine($"Response system message: {responseMessage}");

    return responseMessage?? String.Empty;
}

public async Task<string> TopicPlugin(string context)
{
    Console.WriteLine($"Receive user toplic plugin request");

    Kernel kernel = InitializeKernel();

    KernelPlugin conversationSummary = kernel.ImportPluginFromType<ConversationSummaryPlugin>();

    FunctionResult summary = await kernel.InvokeAsync(
        conversationSummary["GetConversationTopics"], new() { ["input"] = context });
    var responseMessage = summary.GetValue<string>();

    Console.WriteLine($"Response system message: {responseMessage}");

    return responseMessage;
}`

**Comment by sophialagerkranspandey** [comment, marked as answer]
Here's a sample that showcases what you're looking for: https://github.com/microsoft/semantic-kernel/blob/main/dotnet/samples/KernelSyntaxExamples/Example43_GetModelResult.cs
